sql/files/defaultdata/r/run

```diff
@@ -0,0 +1,57 @@
+#!/bin/sh
+
+# R compile wrapper-script for 'compile.sh'.
+# See that script for syntax and more info.
+#
+# This script does not actually "compile" the source, but writes a
+# shell script that will function as the executable: when called, it
+# will execute the source with the correct interpreter syntax, thus
+# allowing this interpreted source to be used transparantly as if it
+# was compiled to a standalone binary.
+#
+# NOTICE: when this compiler script is used with USE_CHROOT on, the
+# chroot requires r to be installed
+
+DEST="$1" ; shift
+MEMLIMIT="$1" ; shift
+MAINSOURCE="${ENTRY_POINT:-$1}"
+
+# Report the entry point, so it can be saved, e.g. for later replay:
+if [ -z "$ENTRY_POINT" ]; then
+    echo "Info: detected entry_point: $MAINSOURCE"
+fi
+
+# No easy way to test R scripts for syntax
+
+# Check if entry point is valid
+if [ ! -r "$MAINSOURCE" ]; then
+    echo "Error: main source file '$MAINSOURCE' is not readable" >&2
+    exit 1
+fi
+
+# Enable creating writable directory - needed by R to store temporary files
+# via `tempfile` function - if this is ommited, we get error like:
+# Fatal error: cannot create 'R_TempDir'
+export CREATE_WRITABLE_TEMP_DIR=1
```

Comment:
Sorry, I was not entirely clear: this should not be needed as we do this in testcase_run.sh automatically 😄 .

Reply:
Oh, I do not completely understand (sorry :sweat_smile: ), but does it mean I have to set this variable as env variable inside my container with application instance? I am a little lost in here

Reply:
No so if you want this feature, change `etc/judgehost-config.php` to have `define('CREATE_WRITABLE_TEMP_DIR', true)` instead of `false`. For in a container you can probably mount a modified version of that file in the correct spot.

Reply:
Could possibly this config file read this value from environment? Otherwise I actually need to care about source code, mounting and so on, which scares me little bit more than simply putting one more env variable inside my docker-compose.yml and rely on docker image entirely :sweat_smile:

Reply:
Let me create a separate issue in `domjudge-packaging` for this, ok? I think that makes the most sense.

Reply:
I mean inside my docker-compose.yml in my own configuration :smile:  here I would only like something like (sorry, this is rather pseudocode, I don't know php):

`define('CREATE_WRITABLE_TEMP_DIR', getenv('CREATE_WRITABLE_TEMP_DIR') ? true : false);`

in `etc/judgehost-config.php`, would it make sense?

Reply:
@eldering what do you think about this change? I think it's fine but you might disagree.

Reply:
Actually, do we need to have this in config at all? I realized with this change included we would read env only to put it back again :see_no_evil:

Reply:
But not everyone can/will set up env variables, for example when not running inside docker but using a shell script or systemd, so it's better to have it.

Reply:
> @eldering what do you think about this change? I think it's fine but you might disagree.

Sounds ok to me. I would prefix the environment variable with `DJ_` or `DOMJUDGE_` to prevent any possible name clash.

Reply:
Sounds good. I can change it on wednesday but if @vicrac wants do to it before go ahead

Reply:
Is this ok now?

Reply:
I’d say only the one in judgehost-config should read DOMJUDGE_, as the other one is in a controlled environment already